### DISCUSSION TITLE
How to grant permissions for each row in Serenity that can be checked for separate users in the Edit permission?

### DISCUSSION THREAD

**Discussion by mvmshakkir** [serenity-is/Serenity#7302] [Help]
How can I set permissions for each row in Serenity so that they can be checked individually for separate users in the edit permissions? For example, I have multiple reports in a module called 'Report Design'. Some reports should be accessible to one user, while others should not be visible to them.

**Comment by LSNicholls** [comment]
I have done this a lot, but it's not possible to explain specifically how unless one knows how the determination of rights is arranged in your data model.  The way I do it in one application for ReportRepository rows happens to be completely different from how I do it for Project rows in the same application.

Generally, though, I'd say you'll easily do it in the PrepareQuery method of the ListHandler for the entity.

I'll try to be more specific if you explain how your rights or permissions need to be evaluated.
